hubspot-style/src/main/java/com/hubspot/immutables/utils/WireSafeEnum.java

```diff
@@ -98,7 +105,7 @@ public static <T extends Enum<T>> WireSafeEnum<T> fromJson(
     ensureJsonCacheInitialized(enumType);
     WireSafeEnum<?> cached = JSON_LOOKUP_CACHE.get(enumType).get(jsonValue);
     if (cached == null) {
-      return new WireSafeEnum<>(enumType, jsonValue);
+      return creator.apply(enumType, jsonValue);
```

Comment:
Thinking about the pros/cons of adding the newly created enum here - pros: speed cons: could be an unbounded set of inputs that map to existing enum values so the cache could get quite large. I think we could split the difference with some bounded LRU cache for entries that were dynamically created but it probably isn't worth thinking about for the first go around on this improvement

Reply:
In this case we're already in a JSON deserialization codepath, so I don't think avoiding Jackson is critical. Obviously it's a nice-to-have, but presumably this is happening in the context of a larger object deserialization and should dwarf the overhead of delegating to Jackson